src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -3519,7 +3516,7 @@ public MethodHandle unreflectConstructor(Constructor<?> c) throws IllegalAccessE
          * @throws IllegalAccessException if access checking fails
          * @throws NullPointerException if the argument is null
          */
-        public MethodHandle unreflectDeconstructor(Deconstructor<?> d) throws IllegalAccessException {
+        public MethodHandle unreflectDeconstructor(MemberPattern<?> d) throws IllegalAccessException {
```

Comment:
Maybe rename this to `unreflectPattern`? If you think future member patterns won't be able to share the same factory as deconstructor, prefer keeping `Deconstructor` signature here.